### PR TITLE
feat: add open ai key to database

### PR DESCRIPTION
**Pull request by LucasXu0**

### CODE REVIEW

File: frontend/app_flowy/assets/translations/en.json [hunk 1]
```diff
@@ -191,6 +191,12 @@
       "create": "Create",
       "folderPath": "Path to store your folder",
       "locationCannotBeEmpty": "Path cannot be empty"
+    },
+    "user": {
+      "name": "Name",
+      "icon": "Icon",
+      "selectAnIcon": "Select an Icon",
```
Comment (appflowy) [inline frontend/app_flowy/assets/translations/en.json:198]: I think Select an icon would be better

File: frontend/app_flowy/assets/translations/en.json [hunk 2]
```diff
@@ -191,6 +191,12 @@
...
+      "pleaseInputYourOpenaiKey": "please input your openai key"
```
Comment (appflowy) [inline frontend/app_flowy/assets/translations/en.json:199]: OpenAI

File: shared-lib/user-model/src/parser/user_openai_key.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+use crate::errors::UserErrorCode;
+
+#[derive(Debug)]
+pub struct UserOpenaiKey(pub String);
+
+impl UserOpenaiKey {
+  pub fn parse(s: String) -> Result<UserOpenaiKey, UserErrorCode> {
+    Ok(Self(s))
```
Comment (appflowy) [inline shared-lib/user-model/src/parser/user_openai_key.rs:8]: Should not be empty string

Comment (LucasXu0) [reply shared-lib/user-model/src/parser/user_openai_key.rs:8]: The OpenAI key can be an empty string.

**Comment by LucasXu0** [comment]
On the dart side, I rename the `openai` to `openAI`. On the rust side, I  rename the `openai` to `openAI` if the code is camel-case style, and keep `openai` if the code is underscore case style.

File: shared-lib/user-model/src/parser/user_openai_key.rs (hunk 3 above)
Comment (appflowy) [reply shared-lib/user-model/src/parser/user_openai_key.rs:8]: Wow, interesting

**Pull request merged**